packages/benchmarks/lit-element/list/index.ts

```diff
@@ -0,0 +1,262 @@
+import {
+  html,
+  LitElement,
+  css,
+  PropertyDeclaration,
+  PropertyValues,
+  UpdatingElement,
+} from 'lit-element';
+import {property, customElement} from 'lit-element/lib/decorators.js';
+
+// IE doesn't support URLSearchParams
+const params = document.location.search
+  .slice(1)
+  .split('&')
+  .map((p) => p.split('='))
+  .reduce((p: {[key: string]: any}, [k, v]) => ((p[k] = v || true), p), {});
+
+type SimpleItem = {[index: string]: string};
+
+function makeItem(prefix: number) {
+  let o: SimpleItem = {};
+  for (let i = 0; i < 99; i++) {
+    o['value' + i] = prefix + ': ' + i;
+  }
+  return o;
+}
+
+function generateData(count: number) {
+  let data = [];
+  for (let i = 0; i < count; i++) {
+    data.push(makeItem(i));
+  }
+  return data;
+}
+
+const data = generateData(250);
+const otherData = generateData(500).slice(250);
+
+const propertyOptions: PropertyDeclaration = {};
+
+let updates: Promise<unknown>[] = [];
```

Comment:
We should figure out a more general pattern that works without this registering of promises. In the Shack benchmark @aomarks had just use a rAF to measure the end of rendering: https://github.com/Polymer/benchmarks/blob/master/lit-element/shack/index-2.js#L21

because the rAF callback goes before style and layout, it was a pretty good measure. If the benchmark takes longer than 16ms, the the callback seemed like it was firing as soon as possible.

Reply:
Done.